roscala/src/main/scala/coop/rchain/rosette/Location.scala

```diff
@@ -78,6 +80,30 @@ object Location {
           .transformS[Ctxt](_.env, (ctxt, env) => ctxt.copy(env = env))
 
       // TODO:
-      case _ => State.pure[Ctxt, StoreResult](Failure)
+      case _ => pure(Failure)
+    }
+  }
+
+  def valWrt[E[_]](loc: Location, v: Ob)(implicit E: MonadError[E, RblError]) =
+    ReaderT[E, TblObject, Ob] { globalEnv =>
```

Comment:
Changing `TblObject` to `GlobalEnv` which is a type alias for `TblObject` should make this easier to read.